src/internal/connector/onedrive/collections.go

```diff
@@ -199,36 +198,35 @@ func deserializeMetadata(
 				// layers would have caught this. Worst case if we don't handle this
 				// we end up in a situation where we're sourcing items from the wrong
 				// base in kopia wrapper.
-				if errors.Is(err, errExistingMapping) {
-					return nil, nil, clues.Wrap(err, "deserializing metadata file").WithClues(ictx)
+				if err != nil {
```

Comment:
We should either keep the old behaviour of failing only for `errExistingMapping` or update the comment to reflect this change.